### DISCUSSION TITLE
performance with files.ignoreUnknown vs files.include

### DISCUSSION THREAD

**Discussion by ianzone** [biomejs/biome#2546] [Q&A] [answered]
if I only set `files.ignoreUnknown` to true, will biome read file contents or just file extensions? Does it have the same performance as I include all supported extensions in `files.include`?

**Comment by ematipico** [comment, marked as answer]
Biome only reads contents only if for files that should be handled.

If a file is unknown, Biome will just emit a diagnostic and move on, without reading its contents.

Same for `files.ignore`

**Reply by ianzone** [reply]
thx for your reply, could you also explain why not set `files.ignoreUnknown` default to true? Since people tend to use biome to check only supported files.

**Reply by ematipico** [reply]
Because there are still cases where it's possible to provide unsupported files e.g. stdin, LSP, git hooks

Also, the flag was added later on, and having set it to true by default might have broken people's code.
